compose/ui/ui/src/desktopMain/kotlin/androidx/compose/ui/awt/ComposeLayer.desktop.kt

```diff
@@ -384,16 +289,108 @@ internal class ComposeLayer(
 
     private var _initContent: (() -> Unit)? = null
 
-    private fun initContent() {
-        if (_component.isDisplayable) {
+    protected fun initContent() {
+        if (component.isDisplayable) {
             _initContent?.invoke()
             _initContent = null
         }
     }
+
+    private fun setCurrentKeyboardModifiers(modifiers: PointerKeyboardModifiers) {
+        platform.windowInfo.keyboardModifiers = modifiers
+    }
+
+    protected fun updateSceneSize() {
+        scene.constraints = Constraints(
+            maxWidth = (component.width * component.density.density).toInt()
+                .coerceAtLeast(0),
+            maxHeight = (component.height * component.density.density).toInt()
+                .coerceAtLeast(0)
+        )
+    }
+
+    protected fun resetSceneDensity() {
+        if (scene.density != component.density) {
+            scene.density = component.density
+            updateSceneSize()
+        }
+    }
+
+    protected fun updateWindowState(window: Window?) {
```

Comment:
Maybe `setParentWindow` instead?

Reply:
What do you mean? I don't know this method anywhere

Reply:
I meant to rename `updateWindowState` to `setParentWindow`.